src/libraries/System.Memory/src/System/Buffers/SequenceReaderExtensions.Binary.cs

```diff
@@ -39,16 +39,15 @@ private static unsafe bool TryReadMultisegment<T>(ref SequenceReader<byte> reade
             Debug.Assert(reader.UnreadSpan.Length < sizeof(T));
 
             // Not enough data in the current segment, try to peek for the data we need.
-            T buffer = default;
-            Span<byte> tempSpan = new Span<byte>(&buffer, sizeof(T));
+            Span<byte> tempSpan = stackalloc byte[sizeof(T)];
```

Comment:
This seems like a regression?
[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9BrIAW2KGIAy2YO258BG6poDMTUgyEMaAbxoMPDANoApXhgBxGC4ZZQAKDABPMRgIADMw3h4ASmSAXXdPL1kAa14jCDBsSVwASS5/DOpPJjtuXGw4mCYUBgBZcgAeABUAPjDk52rPN2Gaj1kxbC5O4EiMGF6GXCmuBgBeZYxsMBziyUKGOYWvXF4AL1iE7vSrcc8AOQgxMJXp5LvPAF9Mj1/vPyBYKhMARaJXRIpdL/Yh1LgNJotdqkHr9Qb/Ub3DzdI4cOJNKAbBgAExgcWwHEkGE+9yeLxCAHdtKtZvNFmEAGTAPEEtDLC4Qm6pGkMH5jIY1LxtGAYHQQYmlcSSMLS2XyxViSQAeTEfAg8JYT3KkiSSQA5tDxWIoLwFNgFkxyEgkXSwpNpqyFks3lxBi4xV8gA===)

Reply:
I am sure whoever wrote the initial code was trying to avoid the stack probe/GS cookie with this workaround.

I am not sure what should we do with code like this, I fear I won't be able to replace 80% of unsafe in dotnet/runtime if we can't accept potential regressions from mandatory security checks like these

Reply:
Current thinking is to re-examine the stack protections in the JIT for .NET 11.

Aside from the concerns above, we are placing increasing amounts of state on the stack that traditionally only lived in the heap.